ast/ddl.go

```diff
@@ -675,7 +677,60 @@ type CreateTableStmt struct {
 
 // Restore implements Node interface.
 func (n *CreateTableStmt) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord("CREATE TABLE ")
+	if n.IfNotExists {
+		ctx.WriteKeyWord("IF NOT EXISTS ")
+	}
+
+	if err := n.Table.Restore(ctx); err != nil {
+		return errors.Annotate(err, "An error occurred while splicing CreateTableStmt Table")
+	}
+	ctx.WritePlain(" ")
+	if n.ReferTable != nil {
+		ctx.WriteKeyWord("LIKE ")
+		if err := n.ReferTable.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while splicing CreateTableStmt ReferTable")
+		}
+	}
+
+	if col := len(n.Cols); col > 0 {
+		ctx.WritePlain("(")
+		for i, col := range n.Cols {
```

Comment:
Consider renaming one of the two `col` variables to avoid confusion.

(Actually `len(n.Cols)` is essentially free to compute I'd suggest removing the outer `col` variable entirely)

Reply:
ok